rust/headless-client/src/linux.rs

```diff
@@ -10,31 +10,62 @@ use connlib_shared::{
 use firezone_cli_utils::setup_global_subscriber;
 use futures::{SinkExt, StreamExt};
 use secrecy::SecretString;
-use std::{
-    future,
-    net::IpAddr,
-    path::{Path, PathBuf},
-    str::FromStr,
-    task::Poll,
-};
+use std::{future, net::IpAddr, path::PathBuf, str::FromStr, task::Poll};
 use tokio::{
     net::{UnixListener, UnixStream},
     signal::unix::SignalKind,
 };
 use tokio_util::codec::LengthDelimitedCodec;
 
+const SOCK_PATH: &str = "/run/firezone-client.sock";
+
 pub async fn run() -> Result<()> {
     let cli = Cli::parse();
     let (layer, _handle) = cli.log_dir.as_deref().map(file_logger::layer).unzip();
     setup_global_subscriber(layer);
 
     match cli.command() {
+        Cmd::Auto => {
+            if let Some(token) = token(&cli).await? {
+                run_standalone(cli, &token).await
```

Comment:
`run_standalone` is already being passed `cli`. Why doesn't it just grab the token from it itself? Would avoid the duplication and make all these functions symmetric by just needing `cli` as an argument.